contracts/auction/LANDAuction.sol

```diff
@@ -210,4 +312,61 @@ contract LANDAuction is Ownable, LANDAuctionStorage {
         }
         return  initialPrice.sub(initialPrice.sub(endPrice).mul(_time).div(duration));
     }
+
+    /**
+    * @dev Convert allowed token to MANA and transfer the change in MANA to the sender
+    * Note that we will use the slippageRate cause it has a 3% buffer
+    * @param _fromToken - ERC20 token to be converted
+    * @param _totalPrice - uint256 of the total amount in MANA
+    * @return bool to confirm the convertion was successfully
+    */
+    function convertSafe(ERC20 _fromToken, uint256 _totalPrice) internal returns (bool) {
+        uint256 prevBalance = manaToken.balanceOf(address(this));
+
+        uint256 tokenRate;
+        (, tokenRate) = dex.getExpectedRate(manaToken, _fromToken, _totalPrice);
+
+        uint256 totalPriceInToken = _totalPrice.mul(tokenRate).div(10 ** 18);
```

Comment:
Maybe it's a good idea to have this as a function for the frontend to query the `totalPriceInToken` and show it to the user?

Reply:
Have did something related [here](https://github.com/decentraland/land-auction/pull/15/files#diff-ac7aac0ba2190ba3d0ee5cbc786152fbR323)